quodlibet/quodlibet/ext/playorder/shufflebygrouping.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Phoenix Dailey
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+import random
+
+from gi.repository import Gtk, GLib
+
+from quodlibet import _
+from quodlibet import app
+from quodlibet.order import OrderInOrder
+from quodlibet.order import OrderRemembered
+from quodlibet.order.reorder import Reorder
+from quodlibet.plugins import PluginConfig
+from quodlibet.plugins.playorder import ShufflePlugin
+from quodlibet.qltk import Icons
+
+
+pconfig = PluginConfig("shufflebygrouping")
+pconfig.defaults.set("grouping", "~grouping~album~albumartist")
+pconfig.defaults.set("grouping_test", "grouping")
```

Comment:
perhaps `grouping_filter` is a clearer name (though I might have misunderstood this)?

Reply:
`grouping_test` was the variable name I was most displeased with. Its purpose is to see if grouping should be applied to a track. At first I just checked if the `grouping` tag was defined, but this had problems when the `grouping` tag was a tied tag (which is useful to distinguish different recordings of the same work). I agree that `grouping_filter` is a better name.